response_operations_ui/views/messages.py

```diff
@@ -130,6 +130,12 @@ def view_conversation(thread_id):
                         form, thread_id=refined_thread[0]["thread_id"], category=thread_conversation["category"]
                     )
                 )
+                # Required for _view_select_survey to know what inbox to correctly redirect to
```

Comment:
All of this session stuff isn't the best, this extends a muddle, where we should just pass cleanly. It can stay, but we really need a story which removes setting session variable like this, as this gets ugly if people jump around the page by url

Reply:
This has been updated.